addons/loyalty/views/loyalty_program_views.xml

```diff
@@ -20,81 +20,125 @@
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'loyalty')]}" colspan="2">
+                                            When customers make an order, they accumulate points they can exchange for rewards on the current order or on a future one.
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'gift_card')]}" colspan="2">
+                                            Gift Cards are created manually or automatically sent by email when the customer orders a gift card product.
+                                            <br/>
+                                            Then, Gift Cards can be used to pay orders.
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'promotion')]}" colspan="2">
+                                            Set up conditional rules on the order that will give access to rewards for customers                                            
+                                            <div groups="base.group_no_one">
+                                                Each rule can grant points to the customer he will be able to exchange against rewards
+                                            </div>
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'ewallet')]}" colspan="2">
+                                            eWallets are created manually or automatically when the customer orders a eWallet product.
+                                            <br/>
+                                            Then, eWallets are proposed during the checkout, to pay orders.
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'promo_code')]}" colspan="2">
+                                            Define Discount codes on conditional rules then share it with your customers for rewards.
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'buy_two_get_one')]}" colspan="2">
+                                            Grant 1 credit for each item bought then reward the customer with Y items in exchange of X credits.
+                                        </p>
+                                        <p class="text-muted" attrs="{'invisible': [('program_type', '!=', 'next_order_coupons')]}" colspan="2">
+                                            Drive repeat purchases by sending a unique, single-use coupon code for the next purchase when a customer buys something in your store.
+                                        </p>
```

Comment:
All these could be a computed field instead, to improve the readability of the view.